pkg/console/service/service_test.go

```diff
@@ -141,87 +147,127 @@ var _ = Describe("Service", func() {
+			extraDataNew    = "data-new-value"
 		)
 
-		apiClient.Fake.PrependReactor("create", "tokenreviews", func(action k8stesting.Action) (bool, runtime.Object, error) {
-			createAction := action.(k8stesting.CreateAction)
-			tokenReview := createAction.GetObject().(*authnv1.TokenReview)
-			tokenReview.Status.Authenticated = true
-			tokenReview.Status.User = authnv1.UserInfo{
-				Username: userName,
-				UID:      userUid,
-				Groups:   groups,
-			}
-			return true, tokenReview, nil
-		})
+		fakeAuthConfig.GetUserHeadersFunc = func() ([]string, error) {
+			return []string{testUserHeader}, nil
+		}
+		fakeAuthConfig.GetGroupHeadersFunc = func() ([]string, error) {
+			return []string{testGroupHeader}, nil
+		}
+		fakeAuthConfig.GetExtraHeaderPrefixesFunc = func() ([]string, error) {
+			return []string{testExtraHeaderPrefix}, nil
+		}
+
+		request.Request.Header = http.Header{}
+		request.Request.Header.Set(testUserHeader, username)
+		request.Request.Header.Set(testGroupHeader, group)
+		request.Request.Header.Set(testExtraHeaderPrefix+extraDataOldKey, extraDataOld)
+		request.Request.Header.Set(testExtraHeaderPrefix+extraDataNewKey, extraDataNew)
 
 		apiClient.Fake.PrependReactor("create", "subjectaccessreviews", func(action k8stesting.Action) (bool, runtime.Object, error) {
 			createAction := action.(k8stesting.CreateAction)
 			sar := createAction.GetObject().(*authzv1.SubjectAccessReview)
 
-			Expect(sar.Spec.User).To(Equal(userName))
-			Expect(sar.Spec.UID).To(Equal(userUid))
-			Expect(sar.Spec.Groups).To(Equal(groups))
+			Expect(sar.Spec.User).To(Equal(username))
+			Expect(sar.Spec.Groups).To(ContainElement(group))
+			Expect(sar.Spec.Extra).To(HaveKeyWithValue(extraDataOldKey, authzv1.ExtraValue{extraDataOld}))
+			Expect(sar.Spec.Extra).To(HaveKeyWithValue(extraDataNewKey, authzv1.ExtraValue{extraDataNew}))
 
 			sar.Status.Allowed = true
 			return true, sar, nil
 		})
 
 		testService.TokenHandler(request, response)
+
+		Expect(recorder.Code).To(Equal(http.StatusOK))
+	})
+
+	It("should fail if auth config returns error", func() {
+		fakeAuthConfig.GetUserHeadersFunc = func() ([]string, error) {
+			return nil, fmt.Errorf("error getting user headers")
+		}
+
+		testService.TokenHandler(request, response)
+
+		Expect(recorder.Code).To(Equal(http.StatusUnauthorized))
+	})
+
+	It("should fail if user does not have permission to access virtualmachineinstances/vnc", func() {
+		apiClient.Fake.PrependReactor("create", "subjectaccessreviews", func(action k8stesting.Action) (bool, runtime.Object, error) {
+			createAction := action.(k8stesting.CreateAction)
+			sar := createAction.GetObject().(*authzv1.SubjectAccessReview)
+			sar.Status.Allowed = false
+			return true, sar, nil
+		})
+
+		testService.TokenHandler(request, response)
+
+		Expect(recorder.Code).To(Equal(http.StatusUnauthorized))
+		Expect(recorder.Body.String()).To(ContainSubstring("does not have permission to access virtualmachineinstances/vnc endpoint"))
```

Comment:
Does it still return this error message? I thought we removed all err messages when returing 401?

Reply:
Yes, the PR to remove the message was not merged yet: https://github.com/kubevirt/vm-console-proxy/pull/23